pkg/ifaces/watcher.go

```diff
@@ -16,44 +21,58 @@ type Watcher struct {
 	interfaces func() ([]Interface, error)
 	// linkSubscriber abstracts netlink.LinkSubscribe implementation, allowing the injection of
 	// mocks for unit testing
-	linkSubscriber func(ch chan<- netlink.LinkUpdate, done <-chan struct{}) error
+	linkSubscriberAt func(ns netns.NsHandle, ch chan<- netlink.LinkUpdate, done <-chan struct{}) error
+	mutex            *sync.Mutex
 }
 
 func NewWatcher(bufLen int) *Watcher {
 	return &Watcher{
-		bufLen:         bufLen,
-		current:        map[Interface]struct{}{},
-		interfaces:     netInterfaces,
-		linkSubscriber: netlink.LinkSubscribe,
+		bufLen:           bufLen,
+		current:          map[Interface]struct{}{},
+		interfaces:       netInterfaces,
+		linkSubscriberAt: netlink.LinkSubscribeAt,
+		mutex:            &sync.Mutex{},
 	}
 }
 
 func (w *Watcher) Subscribe(ctx context.Context) (<-chan Event, error) {
 	out := make(chan Event, w.bufLen)
 
-	go w.sendUpdates(ctx, out)
+	nsHandles, err := getNetNSHandles(ctx)
+	if err != nil {
+		go w.sendUpdates(ctx, 0, out)
```

Comment:
I think this should be:
```suggestion
		go w.sendUpdates(ctx, netns.None(), out)
```

`netns.None()` gives -1, not 0

Reply:
correct fixed it